installer.nsi

```diff
@@ -0,0 +1,223 @@
+; NSIS installer script for open-in-mpv
+; This installer will:
+; - Install the PyInstaller-built EXE
+; - Download and install mpv.exe
+; - Install native messaging JSON files for detected browsers
+
+!include "MUI2.nsh"
+!include "FileFunc.nsh"
+!include "LogicLib.nsh"
+
+; Application information
+!define APP_NAME "open-in-mpv"
+!define APP_VERSION "0.1.3"
+!define APP_PUBLISHER "Andrew Udvare"
+!define APP_URL "https://github.com/Tatsh/${APP_NAME}"
+!define APP_EXEC "${APP_NAME}.exe"
+!define MPV_VERSION "20251214-git-f7be2ee"
+!define MPV_DOWNLOAD_URL "https://downloads.sourceforge.net/project/mpv-player-windows/64bit/mpv-x86_64-${MPV_VERSION}.7z"
+!define MPV_INSTALL_URL "https://mpv.io/installation/"
+
+; Determine current year
+!define /date CURRENT_YEAR "%Y"
+
+; Installer configuration
+Name "${APP_NAME} ${APP_VERSION}"
+OutFile "${APP_NAME}-${APP_VERSION}-setup.exe"
+InstallDir "$LOCALAPPDATA\${APP_NAME}"
+RequestExecutionLevel user
+
+; Interface configuration
+!define MUI_ABORTWARNING
+!define MUI_ICON "${NSISDIR}\Contrib\Graphics\Icons\modern-install.ico"
+!define MUI_UNICON "${NSISDIR}\Contrib\Graphics\Icons\modern-uninstall.ico"
+
+; Pages
+!insertmacro MUI_PAGE_WELCOME
+!insertmacro MUI_PAGE_LICENSE "LICENSE.txt"
+!insertmacro MUI_PAGE_INSTFILES
+!insertmacro MUI_PAGE_FINISH
+
+!insertmacro MUI_UNPAGE_CONFIRM
+!insertmacro MUI_UNPAGE_INSTFILES
+!insertmacro MUI_UNPAGE_FINISH
+
+!insertmacro MUI_LANGUAGE "English"
+
+; Version information
+VIProductVersion "0.1.3.0"
+VIAddVersionKey "ProductName" "${APP_NAME}"
+VIAddVersionKey "CompanyName" "${APP_PUBLISHER}"
+VIAddVersionKey "FileDescription" "${APP_NAME} Installer"
+VIAddVersionKey "FileVersion" "${APP_VERSION}"
+VIAddVersionKey "LegalCopyright" "Copyright (c) ${CURRENT_YEAR} ${APP_PUBLISHER}."
+
+Section "Install" SecInstall
+  SetOutPath "$INSTDIR"
+
+  ; Install the main executable
+  File "dist\${APP_EXEC}"
+
+  ; Check if mpv.exe already exists
+  DetailPrint "Checking for mpv.exe..."
+  IfFileExists "$INSTDIR\mpv.exe" mpv_exists mpv_not_found
+
+  mpv_exists:
+    DetailPrint "mpv.exe already present."
+    Goto mpv_done
+
+  mpv_not_found:
+    ; Download mpv immediately
+    DetailPrint "Downloading mpv from ${MPV_DOWNLOAD_URL}..."
+    NSISdl::download /TIMEOUT=30000 "${MPV_DOWNLOAD_URL}" "$TEMP\mpv.7z"
+    Pop $0
+    ${If} $0 == "success"
+      DetailPrint "Extracting mpv files..."
+      ; Use Nsis7z plugin to extract to temp directory
+      SetOutPath "$TEMP"
+      Nsis7z::ExtractWithDetails "$TEMP\mpv.7z" "Extracting mpv files %s..."
```

Comment:
For this to work, the plugin file https://nsis.sourceforge.io/mediawiki/images/6/69/Nsis7z_19.00.7z must be downloaded and installed into the NSIS Plugins dir, which should be done in the workflow.

See https://nsis.sourceforge.io/How_can_I_install_a_plugin%3F . Must use `Plugins\x64-unicode\nsis7z.dll`.